microsoft-edge/devtools-guide-chromium/memory-problems/heap-snapshot-schema.md

```diff
@@ -0,0 +1,209 @@
+Sliced string | The result of a substring operation, such as using String.prototype.substr or String.prototype.substring. V8 avoids copying string data by instead creating a SlicedString which points to the original string and specifies the start index and length. From a JS perspective, it acts just like any other string, but from a memory profiling perspective, it is different.
+Array | Various internal lists, which show up in the devtools with the category name "(array)". Like Hidden, this category clumps together a bunch of unrelated things. Many of the objects here are named "(object properties)" or "(object elements)", indicating that they contain the string-keyed or numeric-keyed properties of a JavaScript object.
+Code | Things that grow proportionally to the amount of script, and/or the number of times that functions run..  
+Synthetic | Synthetic nodes don't correspond to anything actually allocated in memory. As far as I can tell, these are used only to distinguish the different kinds of GC roots.
+
+
+## Edges
+
+Similar to the **Nodes** array, edges contains all the individual elements needed to re-construct the edges and we can calculate the number of elements of this array by multiplying `edge_count` * `metadata.edge_fields.length`.
+
+Edges are stored sequentially and to read them properly you need to parse the nodes first, because each node knows how many edges it has. 
+
+To re-construct an edge we need 3 elements of information, and we can use `metadata.edge_types` to understand what each element is. 
+
+For example, if you read the first node and its property edge_count is set to 4, that means that the first 4 triplets of the edges array are the elements needed to from this node to some other nodes, etc.
+
+
+`edge_fields` | Description  
+:- | :- 
+type | See `edge_types[0]` to find out what are the possible types
+name_or_index | This could be a number or a string. If it’s a number, it indicates the index of the strings array where the name is stored.
+to_node | Index in the array of the node that it points to
+
+
+`edge_types` | Description  
+:- | :- 
+Internal | Edges that don't correspond to JS-visible names but are still important. As an example, Function instances have a "context" representing the state of variables that were in scope where the function was defined. There is no way for JS code to directly read the "context" of a function, but it's still a very important link when trying to understand retainers.
+Weak | Weak edges don't keep the target alive, and thus are omitted from the Retainers view. Any object with only weak edges pointing to it can be discarded by the GC.
+Hidden | Just like Internal, except these edges don't have nice names because nobody bothered to write special-case code for them.  They're just numbered in increasing order.
```

Comment:
The "nobody bothered to write..." part feels a bit harsh to me :) 
Maybe something like "Like Internal, except these edges don't have unique names and are just numbered in increasing order".

Reply:
woops, that came from an internal email from Seth, we shouldnt use that language in public